src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -319,6 +319,24 @@ private TypeSymbol DecodeTypeOrThrow(ref BlobReader ppSig, SignatureTypeCode typ
                     typeSymbol = DecodeGenericTypeInstanceOrThrow(ref ppSig, out refersToNoPiaLocalType);
                     break;
 
+                case SignatureTypeCode.FunctionPointer:
+                    var signatureHeader = ppSig.ReadSignatureHeader();
+                    var parameters = DecodeSignatureParametersOrThrow(ref ppSig, signatureHeader, typeParameterCount: out int typeParamCount, shouldProcessAllBytes: false);
+                    Debug.Assert(typeParamCount == 0);
```

Comment:
>Debug.Assert(typeParamCount == 0); [](start = 20, length = 34)

I think we should do a check and ```throw new UnsupportedSignatureContent()``` otherwise.